src/ConstructiveSolidGeometry/SurfacePrimitives/Polygon.jl

```diff
@@ -0,0 +1,99 @@
+
+struct Polygon{N,T} <: AbstractPlanarSurfacePrimitive{T}
+    points::SVector{N, CartesianPoint{T}} 
+end
+const Triangle{T} = Polygon{3, T}
+const Quadrangle{T} = Polygon{4, T}
+
+normal(p::Polygon) = normalize((p.points[2] - p.points[1]) × (p.points[3] - p.points[1]))
+
+vertices(p::Polygon) = p.points
+
+extreme_points(p::Polygon) = p.points
```

Comment:
does this primitive also need an `extremum()` method?

Reply:
I don't think so. In general we only need the `extreme_points` method. 
The `extremum` method is only used to generate `extreme_points` for any primitive. 
This can be optimized for each primitive. 
See: https://github.com/lmh91/SolidStateDetectors.jl/blob/8a1c8e377b5a18d7aab349df648e7bd19972b5b6/src/ConstructiveSolidGeometry/ConstructiveSolidGeometry.jl#L72